tests/integ/sagemaker/jumpstart/private_hub/model/test_jumpstart_private_hub_model.py

```diff
@@ -0,0 +1,174 @@
+
+    model_id = "catboost-classification-model"
+
+    model = JumpStartModel(
+        model_id=model_id,
+        role=get_sm_session().get_caller_identity_arn(),
+        sagemaker_session=get_sm_session(),
+        hub_name=os.environ[ENV_VAR_JUMPSTART_SDK_TEST_HUB_NAME],
+    )
+
+    # uses ml.m5.4xlarge instance
+    model.deploy(
+        tags=[{"Key": JUMPSTART_TAG, "Value": os.environ[ENV_VAR_JUMPSTART_SDK_TEST_SUITE_ID]}],
+    )
+
+
+def test_jumpstart_hub_gated_model(setup, add_models):
+
+    model_id = "meta-textgeneration-llama-3-2-1b"
+
+    model = JumpStartModel(
+        model_id=model_id,
+        role=get_sm_session().get_caller_identity_arn(),
+        sagemaker_session=get_sm_session(),
+        hub_name=os.environ[ENV_VAR_JUMPSTART_SDK_TEST_HUB_NAME],
+    )
+
+    # uses ml.g6.xlarge instance
+    predictor = model.deploy(
+        accept_eula=True,
+        tags=[{"Key": JUMPSTART_TAG, "Value": os.environ[ENV_VAR_JUMPSTART_SDK_TEST_SUITE_ID]}],
+    )
+
+    payload = {
+        "inputs": "some-payload",
+        "parameters": {"max_new_tokens": 256, "top_p": 0.9, "temperature": 0.6},
+    }
+
+    response = predictor.predict(payload, custom_attributes="accept_eula=true")
+
+    assert response is not None
+
+
+def test_jumpstart_gated_model_inference_component_enabled(setup, add_models):
+
+    model_id = "meta-textgeneration-llama-2-7b"
+
+    hub_name = os.environ[ENV_VAR_JUMPSTART_SDK_TEST_HUB_NAME]
+
+    region = tests.integ.test_region()
+
+    sagemaker_session = get_sm_session()
+
+    hub_arn = generate_hub_arn_for_init_kwargs(
+        hub_name=hub_name, region=region, session=sagemaker_session
+    )
+
+    model = JumpStartModel(
+        model_id=model_id,
+        role=get_sm_session().get_caller_identity_arn(),
+        sagemaker_session=sagemaker_session,
+        hub_name=os.environ[ENV_VAR_JUMPSTART_SDK_TEST_HUB_NAME],
+    )
+
+    # uses ml.g5.2xlarge instance
+    model.deploy(
+        tags=[{"Key": JUMPSTART_TAG, "Value": os.environ[ENV_VAR_JUMPSTART_SDK_TEST_SUITE_ID]}],
+        accept_eula=True,
+        endpoint_type=EndpointType.INFERENCE_COMPONENT_BASED,
+    )
+
+    predictor = retrieve_default(
+        endpoint_name=model.endpoint_name,
+        sagemaker_session=sagemaker_session,
+        tolerate_vulnerable_model=True,
+        hub_arn=hub_arn,
+    )
+
+    payload = {
+        "inputs": "some-payload",
+        "parameters": {"max_new_tokens": 256, "top_p": 0.9, "temperature": 0.6},
+    }
+
+    response = predictor.predict(payload)
+
+    assert response is not None
+
+    model = JumpStartModel.attach(
+        predictor.endpoint_name, sagemaker_session=sagemaker_session, hub_name=hub_name
+    )
+    assert model.model_id == model_id
+    assert model.endpoint_name == predictor.endpoint_name
+    assert model.inference_component_name == predictor.component_name
+
+
+def test_instatiating_model(setup, add_models):
```

Comment:
i wonder if we can run all the tests currently for JumpStart but for PrivateHub. 

RIsk is that this test coverage drifts from the non-private-hub tests as new models/features are added.

Reply:
> I wonder if we can run all the tests currently for JumpStart but for PrivateHub.
I am not sure, are you thinking about reusing the test code of Jumpstart public hub specific tests and integrate PrivateHub workflow triggers there? won't it be too complicated? keeping separate increases the readability? also not all features available for content type Model is available for content type ModelReferences?

> RIsk is that this test coverage drifts from the non-private-hub tests as new models/features are added.
sorry I can't understand what you mean by this.

Reply:
I think Evan is suggesting defining a single set of model tests, and systematically test them against the public hub and a private hub. That would be substantial rework from your current PR though.

Reply:
fix typo please: `test_instantiating_model`

Reply:
> I think Evan is suggesting defining a single set of model tests, and systematically test them against the public hub and a private hub. That would be substantial rework from your current PR though.

+1 That would require more efforts and rework. I can take it as an improvement but my question is I want to understand what's the benefit of it? is it better design or just a personal preference to implement it?